kv/kvserver: TestReplicaRangefeed failed

[(kv/kvserver).TestReplicaRangefeed failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2199075&tab=buildLog) on [master@c9e3f980976caeee743355291ca795f132fac618](https://github.com/cockroachdb/cockroach/commits/c9e3f980976caeee743355291ca795f132fac618):

```
=== RUN   TestReplicaRangefeed
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaRangefeed490412297
--- FAIL: TestReplicaRangefeed (2.91s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaRangefeed490412297
    test_log_scope.go:63: use -show-logs to present logs inline
    replica_rangefeed_test.go:188: incorrect events on stream 0, found [val:<key:"b" value:<raw_bytes:"^\260\206U\001\022" timestamp:<wall_time:1597877040597844695 logical:1 > > prev_value:<timestamp:<> > >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1597877036257134242 > > ], want [val:<key:"b" value:<raw_bytes:"^\260\206U\001\022" timestamp:<wall_time:1597877040597844695 logical:1 > > prev_value:<timestamp:<> > >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeed PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeed.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeed failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2205817&tab=buildLog) on [master@4ab67e2f9d063bd2da8b53e1d400d967e71db7c7](https://github.com/cockroachdb/cockroach/commits/4ab67e2f9d063bd2da8b53e1d400d967e71db7c7):

```
=== RUN   TestReplicaRangefeed
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaRangefeed118544219
--- FAIL: TestReplicaRangefeed (4.17s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaRangefeed118544219
    test_log_scope.go:63: use -show-logs to present logs inline
    replica_rangefeed_test.go:263: incorrect events on stream 0, found [val:<key:"b" value:<raw_bytes:"^\260\206U\001\022" timestamp:<wall_time:1597984152733965297 logical:1 > > prev_value:<timestamp:<> > >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  val:<key:"c" value:<raw_bytes:"\000\000\000\000\003val2" timestamp:<wall_time:1597984152733965297 logical:2 > > prev_value:<timestamp:<> > >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1597984149282843337 > >  val:<key:"m" value:<raw_bytes:"3\321\334\035\003val3" timestamp:<wall_time:1597984152733965297 logical:3 > > prev_value:<timestamp:<> > >  val:<key:"b" value:<raw_bytes:"\221\n\023\314\001$" timestamp:<wall_time:1597984152733965297 logical:4 > > prev_value:<raw_bytes:"^\260\206U\001\022" timestamp:<> > >  val:<key:"b" value:<raw_bytes:"e\336\246\235\0012" timestamp:<wall_time:1597984152733965297 logical:5 > > prev_value:<raw_bytes:"\221\n\023\314\001$" timestamp:<> > > ], want [val:<key:"b" value:<raw_bytes:"^\260\206U\001\022" timestamp:<wall_time:1597984152733965297 logical:1 > > prev_value:<timestamp:<> > >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  val:<key:"c" value:<raw_bytes:"\000\000\000\000\003val2" timestamp:<wall_time:1597984152733965297 logical:2 > > prev_value:<timestamp:<> > >  val:<key:"m" value:<raw_bytes:"3\321\334\035\003val3" timestamp:<wall_time:1597984152733965297 logical:3 > > prev_value:<timestamp:<> > >  val:<key:"b" value:<raw_bytes:"\221\n\023\314\001$" timestamp:<wall_time:1597984152733965297 logical:4 > > prev_value:<raw_bytes:"^\260\206U\001\022" timestamp:<> > >  val:<key:"b" value:<raw_bytes:"e\336\246\235\0012" timestamp:<wall_time:1597984152733965297 logical:5 > > prev_value:<raw_bytes:"\221\n\023\314\001$" timestamp:<> > > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeed PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeed.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/replica_rangefeed_test.go
+++ b/pkg/kv/kvserver/replica_rangefeed_test.go
@@ -21,7 +21,9 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/closedts"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
@@ -86,11 +88,20 @@ func TestReplicaRangefeed(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
-	tc := testcluster.StartTestCluster(t, 3,
-		base.TestClusterArgs{
-			ReplicationMode: base.ReplicationManual,
-		},
-	)
+	const numNodes = 3
+	args := base.TestClusterArgs{
+		ReplicationMode:   base.ReplicationManual,
+		ServerArgsPerNode: make(map[int]base.TestServerArgs, numNodes),
+	}
+	for i := 0; i < numNodes; i++ {
+		// Disable closed timestamps as this test was designed assuming no closed
+		// timestamps would get propagated.
+		settings := cluster.MakeTestingClusterSettings()
+		closedts.TargetDuration.Override(&settings.SV, 24*time.Hour)
+		kvserver.RangefeedEnabled.Override(&settings.SV, true)
+		args.ServerArgsPerNode[i] = base.TestServerArgs{Settings: settings}
+	}
+	tc := testcluster.StartTestCluster(t, numNodes, args)
 	defer tc.Stopper().Stop(ctx)
 
 	ts := tc.Servers[0]
@@ -140,7 +151,6 @@ func TestReplicaRangefeed(t *testing.T) {
 				Span:     rangefeedSpan,
 				WithDiff: true,
 			}
-			kvserver.RangefeedEnabled.Override(&store.ClusterSettings().SV, true)
 			pErr := store.RangeFeed(&req, stream)
 			streamErrC <- pErr
 		}(i)

